CentralAuthPlugin.php

```diff
@@ -225,21 +226,30 @@ public function filterLoginAdapter($adapter, $args)
                 }
             }
 
+            // Get LDAP bind password from the ini file if configuration
+            // is set up to do this and a bind username has been entered
+            if (isset($options['username']) && $options['use_ini']) {
+                $ldap_ini = parse_ini_file ('ldap.ini', true);
+                if (isset($ldap_ini['ldap']['bind_password'])) {
+                    $options['password'] = $ldap_ini['ldap']['bind_password'];
+                }
+            }
+
+            // Remove $options['use_ini'] because it's not an option that
+            // is recognized by Zend_Ldap, it's only use is to tell us
+            // where to find the 'password' option
+            if (isset($options['use_ini'])) {
+                unset($options['use_ini']);
+            }
+
             // Create new auth adapter with the options, username and password.
             $adapterLdap = new CentralAuth_LdapAdapter(
                 array('ldap' => $options),
                 $form->getValue('username'),
                 $form->getValue('password')
             );
 
-            // Attempt to authenticate using LDAP.
-            $result = $adapterLdap->authenticate();
```

Comment:
@kloor Here is where the initial Ldap authentication happens, to check whether or not the user can use Ldap, or whether they must use the default Omeka authentication.